chain-impl-mockchain/src/ledger.rs

```diff
@@ -137,113 +138,113 @@ pub mod test {
+        );
+        let tx = Transaction {
+            inputs: vec![utxo0],
+            outputs: vec![Output(user1_address, Value(1))],
+        };
+        let signed_tx = SignedTransaction {
+            transaction: tx,
+            witnesses: vec![],
+        };
+        assert_eq!(
+            Err(Error::NotEnoughSignatures(1, 0)),
+            ledger.diff_transaction(&signed_tx)
+        )
+    }
 
-        #[test]
-        fn cant_loose_money() {
-            use chain_core::property::Ledger;
-            use chain_core::property::Transaction;
-            let (pk1, user1_address) = make_key(0);
-            let tx0_id = TransactionId::hash_bytes(&[0]);
-            let value = Value(42000);
-            let utxo0 = UtxoPointer {
-                transaction_id: tx0_id,
-                output_index: 0,
-                value: value,
-            };
-            let ledger = crate::ledger::Ledger::new(
-                vec![(utxo0, Output(user1_address.clone(), Value(10)))]
-                    .iter()
-                    .cloned()
-                    .collect(),
-            );
-            let output0 = Output(user1_address, Value(9));
-            let tx = crate::transaction::Transaction {
-                inputs: vec![utxo0],
-                outputs: vec![output0],
-            };
-            let witness = Witness::new(&tx.id(), &pk1);
-            let signed_tx = SignedTransaction {
-                transaction: tx,
-                witnesses: vec![witness],
-            };
-            assert_eq!(
-                Err(Error::TransactionSumIsNonZero(10, 9)),
-                ledger.diff_transaction(&signed_tx)
-            )
-        }
+    #[test]
+    pub fn tx_wrong_witness() -> () {
+        use chain_core::property::Ledger;
+        use chain_core::property::Transaction;
+        let mut rng = rand::thread_rng();
+        let (_, user0_address) = make_key(&mut rng);
+        let tx0_id = TransactionId::hash_bytes(&[0]);
+        let value = Value(42000);
+        let utxo0 = UtxoPointer {
+            transaction_id: tx0_id,
+            output_index: 0,
+            value: value,
+        };
+        let ledger = crate::ledger::Ledger::new(
+            vec![(utxo0, Output(user0_address.clone(), value))]
+                .iter()
+                .cloned()
+                .collect(),
+        );
+        let output0 = Output(user0_address, value);
+        let tx = crate::transaction::Transaction {
+            inputs: vec![utxo0],
+            outputs: vec![output0.clone()],
+        };
+        let (pk1, _) = make_key(&mut rng);
+        let witness = Witness::new(&tx.id(), &pk1);
+        let signed_tx = SignedTransaction {
+            transaction: tx,
+            witnesses: vec![witness.clone()],
+        };
+        assert_eq!(
+            Err(Error::InvalidSignature(utxo0, output0, witness)),
+            ledger.diff_transaction(&signed_tx)
+        )
+    }
 
-    */
+    #[test]
+    fn cant_loose_money() {
```

Comment:
Typo: lose. But you can't loose that money, either :stuck_out_tongue_winking_eye: